codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -401,6 +401,15 @@ public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
                                 "Multiple content-length headers received");
                     }
                 }
+            } else {
+                // Need to check trailers don't contain pseudo headers. According to RFC
```

Comment:
"RFC" is now a bit generic for HTTP/2, because they redefined it. Seems you are referring to RFC 7540, not RFC 9113 which says "Pseudo-header fields MUST NOT appear in a trailer section." They are pretty similar, but for new code, better to quote the newer text.

Reply:
Makes sense, I'll update to match the new RFC